pkg/controller/apmserverelasticsearchassociation/apmserverelasticsearchassociation_controller.go

```diff
@@ -406,24 +409,5 @@ func (r *ReconcileApmServerElasticsearchAssociation) reconcileElasticsearchCA(ct
 // now redundant old user object/secret. This function lists all resources that don't match the current name/namespace
 // combinations and deletes them.
 func deleteOrphanedResources(ctx context.Context, c k8s.Client, as *apmv1.ApmServer) error {
```

Comment:
Looks like we could remove this function and call `association.DeleteOrphanedResources(ctx, c, as, NewResourceLabels(as.Name), createdByApmServer) ` directly?

Reply:
I would be happy to do that but there are some unit tests that rely on it _(See my comment in the description)_.